src/SettingsMapper.php

```diff
@@ -45,16 +45,18 @@ public function load(string $settingsClass): Collection
     }
 
     public function save(
-        string $settingsClass,
+        string     $settingsClass,
         Collection $properties
     ): Collection {
         $config = $this->getConfig($settingsClass);
 
         $this->ensureNoMissingSettings($config, $properties, 'saving');
 
-        $changedProperties = $properties
-            ->reject(fn ($payload, string $name) => $config->isLocked($name))
-            ->each(function ($payload, string $name) use ($config) {
+        $propertiesToUpdate = $properties
+            ->reject(fn ($payload, string $name) => $config->isLocked($name));
+
+        (clone $propertiesToUpdate)
```

Comment:
Why would you add a clone here?

Reply:
You're correct that there's no need to use reject, as it returns a new filtered collection.